src/forward.jl

```diff
@@ -108,8 +108,8 @@ function GenericForwardDiffADJtprod(
 )
   return GenericForwardDiffADJtprod()
 end
-function Jtprod!(::GenericForwardDiffADJtprod, Jtv, f, x, v, ::Val)
-  Jtv .= ForwardDiff.gradient(x -> dot(f(x), v), x)
+function Jtprod!(::GenericForwardDiffADJtprod, Jtv::S, f, x::S, v::S, ::Val) where{S}
+  Jtv .= ForwardDiff.gradient(x -> dot(f(x), v), x)::S
```

Comment:
This would be hiding the issues I think. `Jtv` is of type `S`, so we shouldn't have to convert it to `S` again. If that's so, it indicates that there are some issues and it probably should not be done silently.
We then need to either add a warning or/and add a unit test + open an issue to track this.
```suggestion
  Jtv .= ForwardDiff.gradient(x -> dot(f(x), v), x)
```